pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -600,6 +600,13 @@ func ResolvePipelineTask(
 		if run != nil {
 			rpt.RunObject = run
 		}
+	case rpt.PipelineTask.IsMatrixed() && rpt.PipelineTask.Matrix.ContainsWholeArrayResultReferences():
+		// If the matrix contains whole array results references [*], resolve task resources instead of
```

Comment:
I'm a bit confused by this comment. I understand why the number of matrixed pipeline tasks isn't known at this point, but I'm not sure why we need to "resolve task resources" here instead. ("resolving resources" started out referring to pipelineresources, which have since been removed, and the name expanded in scope to resolving params, remote tasks, etc etc.) From reading the code it seems like the main purpose is to fetch the referenced task spec, so maybe the comment just needs a bit of rewording to focus on that?

Reply:
@lbernick Updated this to just get and resolve the singular TaskName